src/test/java/io/managed/services/test/kafka/KafkaMgmtAPITest.java

```diff
@@ -218,6 +218,123 @@ public void testCreateTopics() {
         kafkaInstanceApi.createTopic(metricTopicPayload);
     }
 
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testForbiddenToExceedPartitionLimitByTopicCreation() {
+
+        var topicName = "topic-part-exceed-create";
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        log.info("Max partition limit per given instance: {}", maxPartitionLimit);
+
+        log.info("Attempt to create topic '{}', with too many partitions: {}", topicName, maxPartitionLimit + 1);
+        var payload = new NewTopicInput()
+                .name(topicName)
+                .settings(new TopicSettings().numPartitions(maxPartitionLimit + 1));
+        assertThrows(ApiGenericException.class,
+                () -> kafkaInstanceApi.createTopic(payload));
+    }
+
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testForbiddenToExceedParitionLimitByTopicConfiguration() {
+
+        var topicName = "topic-part-exceed-conf";
+        KafkaInstanceApiUtils.applyTopic(kafkaInstanceApi, topicName);
+
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        assertThrows(ApiGenericException.class,
+                () -> KafkaInstanceApiUtils.updateTopicPartition(kafkaInstanceApi, topicName, maxPartitionLimit + 1));
+    }
+
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testTotalPartitionLimitByTopicCreation() {
+
+        var topicName = "topic-part-total-exceed-create";
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        final int currentPartitionCount = KafkaInstanceApiUtils.getPartitionCountTotal(kafkaInstanceApi);
+
+        var payloadTopicMax = new NewTopicInput()
+                .name(topicName)
+                .settings(new TopicSettings().numPartitions(maxPartitionLimit));
+
+        // I can exceed the max limit with single new topic
+        if (currentPartitionCount > 0) {
+            log.info("Trying to create topic with {} partitions, while total partition limit is {}, and instance already has {} partitions", maxPartitionLimit, maxPartitionLimit, currentPartitionCount);
```

Comment:
I'm not quite sure of intent of this branch?  How is it different to testForbiddenToExceedPartitionLimitByTopicCreation?

Reply:
I probably can rename the `testForbiddenToExceedPartitionLimitByTopicCreation` to  `testForbiddenToExceedPartitionLimitOnTopicCreationSpecification`  as this test makes sure that I cannot breach the limit of partitions inside specific topic, i.e., the limit is 1500 i want to make sure that user cannot create the topic with 6000 partitions in it.

The test in branch does not breach the limit 1500 directly (instead it will breach it only when it sums up with other previously created partitions by already existing topics).

Reply:
@k-wall They're subtly different branches in our CreateTopicPolicy validation: `testForbiddenToExceedPartitionLimitByTopicCreation` would fail [this check](https://github.com/bf2fc6cc711aee1a0c2a/kafka-group-authorizer/blob/f3f77ec742282e4fdfe3067a164aa82e74f9bb23/src/main/java/io/bf2/kafka/topic/ManagedKafkaCreateTopicPolicy.java#L103), whereas I think this branch in this test would end up [here](https://github.com/bf2fc6cc711aee1a0c2a/kafka-group-authorizer/blob/f3f77ec742282e4fdfe3067a164aa82e74f9bb23/src/main/java/io/bf2/kafka/topic/ManagedKafkaCreateTopicPolicy.java#L111).

Whether we need to test every branch like this in e2e tests, versus shifting left to rely on unit/integration tests, is maybe a different question worth thinking about?

Reply:
In general, I favour the shift left approach.  That is, there should be a smaller, judiciously chosen, set of tests as each tier, as you go up the tiers of the abstraction.     However, in this case, as the solution is not ideal, I think covering all the case is best.

Reply:
I'm see not seeing the difference between L233 and L266

Reply:
actually L233 works well in production so I assumed there is different mechanism (probably as you are mentioning Admin level). 
So to finilize this can we go with this way or should I remove the test from L233 ?